src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -311,33 +324,77 @@ public async Task<IAsyncCursor<TResult>> ExecuteAsync(IReadBinding binding, Canc
             }
 
             IAsyncCursor<RawBsonDocument> cursor;
+            IBatchInfo batchInfo;
+            BsonTimestamp initialOperationTime;
             using (var channelSource = await binding.GetReadChannelSourceAsync(cancellationToken).ConfigureAwait(false))
             using (var channel = await channelSource.GetChannelAsync(cancellationToken).ConfigureAwait(false))
             using (var channelBinding = new ChannelReadBinding(channelSource.Server, channel, binding.ReadPreference, binding.Session.Fork()))
             {
-                cursor = await ExecuteAggregateOperationAsync(channelBinding, resuming: false, cancellationToken).ConfigureAwait(false);
-                SaveInitialOperationTimeIfRequired(channel, channelBinding, cursor);
+                cursor = await ExecuteAggregateOperationAsync(channelBinding, cancellationToken).ConfigureAwait(false);
+                batchInfo = GetBatchInfoOrThrow(cursor);
+                initialOperationTime = GetInitialOperationTimeIfRequired(channel, channelBinding, batchInfo);
             }
 
-            return new ChangeStreamCursor<TResult>(cursor, _resultSerializer, bindingHandle.Fork(), this);
+            var postBatchResumeToken = GetInitialPostBatchResumeTokenIfRequired(batchInfo);
+
+            return new ChangeStreamCursor<TResult>(
+                cursor, 
+                _resultSerializer, 
+                bindingHandle.Fork(), 
+                this,
+                postBatchResumeToken, 
+                initialOperationTime,
+                _startAfter,
+                _resumeAfter,
+                _startAtOperationTime);
         }
 
         /// <inheritdoc />
         public IAsyncCursor<RawBsonDocument> Resume(IReadBinding binding, CancellationToken cancellationToken)
         {
-            return ExecuteAggregateOperation(binding, resuming: true, cancellationToken);
+            return ExecuteAggregateOperation(binding, cancellationToken);
         }
 
         /// <inheritdoc />
         public Task<IAsyncCursor<RawBsonDocument>> ResumeAsync(IReadBinding binding, CancellationToken cancellationToken)
         {
-            return ExecuteAggregateOperationAsync(binding, resuming: true, cancellationToken);
+            return ExecuteAggregateOperationAsync(binding, cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public IChangeStreamOperation<TResult> WithResumeOptions(BsonDocument startAfter, BsonDocument resumeAfter, BsonTimestamp startAtOperationTime)
```

Comment:
Let's get rid of this method and instead of cloning the operation for Resume let's just reconfigure the three Resume values and re-use the existing operation.

Reply:
Done